### PR TITLE
Add prop to toggle time display between UTC and local

### LINKED ISSUE #30893: Bugs arising after migrating to D3's scaleUtc utility
Following bugs have been identified/reported:
1. In some cases, axis ticks only display the time component instead of a more meaningful representation.
   Possible reason: The `multiFormat` utility is not working well with `scaleUtc`.

   image

2. Axis ticks are shown in UTC time, whereas other elements of the chart, such as the callout, display local time.

   image

### LINKED ISSUE #31269: [Bug]: Charting: customDateTimeFormatter and timeFormatLocale does not display the correct date for x axis
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (36) x64 Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz
    Memory: 33.23 GB / 63.92 GB
  Browsers:
    Chrome: 124.0.6367.119
    Edge: Chromium (123.0.2420.97), ChromiumDev (126.0.2552.0)
    Internet Explorer: 11.0.19041.3636
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/Gabnight/pen/wvZLroo

## Actual Behavior
When creating point in local time zone (GMT +1 in the example) for line chart or area chart.
Ex:
            {
              x: new Date('2020-03-04T00:00:00.000'),
              xAxisCalloutData: new Date('2020-03-04T00:00:00.000'),
              y: 284000,
            },
The x axis tick is set at 1 am with GMT +1 which make weird UI because the tick is not aligned with the point.
Moreover, the localization of the date is broken because we need a date at midnight to display the day instead of hour on the customDateTimeFormatter callback method.
It seems the date passed to the chart is handled as UTC even if GMT +1 value is specified in it.

Screenshot:
image

## Expected Behavior
point and tick should be aligned at midnight.

Note:
This bug is also visible in the React-charting demo website:
https://developer.microsoft.com/en-us/fluentui#/controls/web/linechart
image

This regression seems to have been brought about by this change:
image
in the PR:
https://github.com/microsoft/fluentui/pull/30315/

### Logs

_No response_

### Requested priority

High

### Products/sites affected

make.powerautomate.com

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by krkshitij**
## Previous Behavior

Axis ticks are displayed in UTC time, while the time shown in the callout uses local time.

## New Behavior

Axis scale, ticks, and the time display in callouts use either UTC or local time, depending on the new `useUTC` prop.

use UTC time
use local time

## Related Issue(s)

- Fixes #30893
- Fixes #31269

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ import {
   domainRangeOfNumericForAreaChart,
```
Comment (fabricteam) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:33]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31525&commitId=a6c7dcecf8f791bb08d41bb95a040565f38cd41e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Pivot - Overflow</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Removed|

</details>


  <details><summary><b>react-charting-LineChart</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Basic - Dark Mode.hover.chromium.png|120| Changed|
|react-charting-LineChart.Basic.hover.chromium.png|149| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/utilities/utilities.ts [hunk 2]
```diff
@@ -1353,3 +1379,8 @@ export const getScalePadding = (prop: number | undefined, shorthandProp?: number
 export const isScalePaddingDefined = (prop: number | undefined, shorthandProp?: number): boolean => {
   return typeof prop === 'number' || typeof shorthandProp === 'number';
 };
+
+export const formatDate = (date: Date, useUTC?: boolean) => {
+  const timeFormat = useUTC ? d3UtcFormat : d3TimeFormat;
+  return timeFormat('%-e %b %Y, %H:%M')(date) + (useUTC ? ' UTC' : '');
```
Comment (AtishayMsft) [inline packages/react-charting/src/utilities/utilities.ts:1385]: use GMT instead of UTC

**Comment by AtishayMsft** [comment]
Can you add in description how the new behavior looks.

**Review by AtishayMsft — APPROVED** [review]

File: packages/react-charting/src/utilities/utilities.ts (hunk 2 above)
Comment (krkshitij) [reply packages/react-charting/src/utilities/utilities.ts]: done

**Pull request merged**